Tweak: Upgrade minimum required PHP version from 7.0 to 7.3 [ED-11983]

rami-elementor opened the pull request:
> By the end of the 3rd quarter, Elementor will raise the minimum required version to PHP 7.3, along with upgrading the recommended version to PHP 7.4.

Source: [Elementor’s Roadmap to Upgrade PHP Versions
](https://developers.elementor.com/elementors-roadmap-to-upgrade-php-versions/)

rami-elementor committed:
Tweak: Upgrade minimum required PHP version from 7.0 to 7.3

yotamselementor commented on readme.txt:
```diff
@@ -147,12 +147,12 @@ There is also a managed security [Bug Bounty program](https://go.elementor.com/w
 * MySQL version 5.0 or greater
 
-= We recommend your host supports: =
+= Recommend Requirements =
```
```suggestion
= Recommended Requirements =
```

rami-elementor committed:
Update readme.txt

Co-authored-by: yotamselementor <83912471+yotamselementor@users.noreply.github.com>

rami-elementor committed:
Update composer

rami-elementor committed:
Merge branch 'php-version' of https://github.com/elementor/elementor into php-version

matipojo commented on .github/workflows/phpunit.yml:
```diff
@@ -36,7 +36,7 @@ jobs:
       matrix:
         wordpress_versions: ['nightly', 'latest', '6.0', '6.1', '6.2']
-        php_versions: ['7.0', '7.1', '7.2', '7.3', '7.4']
+        php_versions: ['7.3', '7.4', '8.0', '8.1', '8.2']
```
```suggestion
        php_versions: ['7.3', '7.4']
```

yotamselementor committed:
Regenerate composer.lock (#23652)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

Co-authored-by: Yotam Shapira <yotams@MacBook-Pro.local>

yotamselementor committed:
Upgrade yoast/phpunit-polyfills (#23655)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

Co-authored-by: Yotam Shapira <yotams@MacBook-Pro.local>

rami-elementor commented:
Hi @yotamselementor , 

@matipojo suggested to spilt all the test-related code changes to a separate task.

rami-elementor committed:
Revert phpunit

rami-elementor committed:
Revert composer

rami-elementor committed:
Revert

yotamselementor commented:
> 

@rami-elementor I don't understand - what other changes do we have here that you want to split out?

arielk committed:
Merge branch 'main' into php-version

TzviRabinovitch committed:
Merge branch 'main' into php-version

arielk committed:
Merge branch 'main' into php-version

StyleShit commented:
what about composer.json?

rami-elementor commented:
@StyleShit see https://github.com/elementor/elementor/pull/23619#issuecomment-1704295003 above
